pymc/data.py

```diff
@@ -165,6 +184,7 @@ def is_valid_observed(v) -> bool:
             and isinstance(v.owner.op.scalar_op, Cast)
             and is_valid_observed(v.owner.inputs[0])
         )
+        or not has_random_ancestor(v)
```

Comment:
The previous branch is now a subset of the new more general behavior

Reply:
Removed